docs/nlu/using-bert.rst

```diff
@@ -0,0 +1,159 @@
+representations for each incoming message(at word and character levels). The heavy configuration replaces it with a
+BERT model inside the pipeline. :ref:`HFTransformersNLP` is a utility component that does the heavy lifting work of loading the
+``BERT`` model in memory. Under the hood it leverages HuggingFace's `Transformers library <https://huggingface.co/transformers/>`_ to initialize the specified language model.
+Notice that we add two additional components :ref:`LanguageModelTokenizer` and :ref:`LanguageModelFeaturizer` which
+pick up the tokens and feature vectors respectively that are constructed by the utility component.
+
+We use the same :ref:`diet-classifier` model for combined intent classification and entity recognition in both cases.
+
+Run the Pipelines
+-----------------
+
+You can run both configurations yourself.
+
+.. code-block:: yaml
+
+    mkdir gridresults
+    rasa test nlu --config configs/config-light.yml \
+                  --cross-validation --runs 1 --folds 2 \
+                  --out gridresults/config-light
+    rasa test nlu --config configs/config-heavy.yml \
+                  --cross-validation --runs 1 --folds 2 \
+                  --out gridresults/config-heavy
+
+When this runs you should see logs appear. We've picked a few
+of those lines to list them here. 
+
+.. code-block:: txt
+
+    # output from the light model
+    2020-03-30 16:21:54 INFO     rasa.nlu.model  - Starting to train component DIETClassifier
+    Epochs: 100%|███████████████████████████████| 50/50 [04:30<00:00, ...]
+    2020-03-30 16:23:53 INFO     rasa.nlu.test  - Running model for predictions:
+    100%|███████████████████████████████████████| 2396/2396 [01:23<00:00, 28.65it/s]
+    ...
+    # output from the heavy model
+    2020-03-30 16:47:04 INFO     rasa.nlu.model  - Starting to train component DIETClassifier
+    Epochs: 100%|███████████████████████████████| 50/50 [04:33<00:00,  ...]
+    2020-03-30 16:49:52 INFO     rasa.nlu.test  - Running model for predictions:
+    100%|███████████████████████████████████████| 2396/2396 [07:20<00:00,  5.69it/s]
+
+From the logs we can gather an important observation. 
+The heavy model is a fair bit slower, not in training, but at inference time
+we see a ~6 fold increase. Depending on your use-case this is 
+something to seriously consider.
+
+Results
+-------
+
+We've summarised the results into two charts, one for intents and
+one for entities.
+
+
+Intent Results 
+~~~~~~~~~~~~~~
+
+.. image:: /_static/images/bert-intents.png
+
+Entity Results 
+~~~~~~~~~~~~~~
+
+.. image:: /_static/images/bert-entities.png
+
+Observations 
+~~~~~~~~~~~~
+
+On all fronts we see that the model with the Bert embeddings performs better. 
+But it deserves mentioning that the effect is more pronounced in the entities.
+Note that these results may not be the same on your use-case. Every assistant 
+is different so it is important that you keep comparing. 
```

Comment:
Can we motivate this point a bit better, making it explicit that there are two trade-offs at play here - 
1. Which task is more important - intent classification or entity recognition?
2. Is accuracy more important or latency of the bot during prediction.

Reply:
I've expanded the points a bit.